maintenance/health/servicehealthcheck/background_healthchecker.go

```diff
@@ -0,0 +1,89 @@
+package servicehealthcheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// BackgroundHealthCheck is an interface that, when implemented, will run the
+// HealthCheck in the background, and cache the result, that will be fetched by the HealthHandler
+type BackgroundHealthCheck interface {
+	HealthCheck
+	Interval() time.Duration
+}
+
+// BackgroundHealthCheckMaxWait allows to use a custom amount of time to wait before failing the health check
+type BackgroundHealthCheckMaxWait interface {
+	MaxWait() time.Duration
+}
+
+// registerBackgroundHealthCheck will run the HealthCheck in the background at every given interval.
+// The returned backgroundStateHealthChecker is then used to query this cached state when needed.
+func registerBackgroundHealthCheck(name string, bhc BackgroundHealthCheck) *backgroundStateHealthChecker {
+	var bgState ConnectionState
+
+	go func() {
```

Comment:
As far as I can see this routine does not have any panic recovery, right? Are we fine with it dying and not reporting any results anymore and then see that because no results are being reported?